src/web/app/pages-instructor/instructor-session-result-page/instructor-session-result-page.component.ts

```diff
@@ -297,24 +298,63 @@ export class InstructorSessionResultPageComponent extends InstructorCommentsComp
       // Do not re-fetch data
       return;
     }
-    this.feedbackSessionsService.getFeedbackSessionResults({
-      questionId,
-      courseId: this.session.courseId,
-      feedbackSessionName: this.session.feedbackSessionName,
-      intent: Intent.FULL_DETAIL,
-    })
-    .subscribe((resp: SessionResults) => {
-      if (resp.questions.length) {
-        const responses: QuestionOutput = resp.questions[0];
-        this.questionsModel[questionId].responses = responses.allResponses;
-        this.questionsModel[questionId].statistics = responses.questionStatistics;
-        this.questionsModel[questionId].hasPopulated = true;
 
-        this.preprocessComments(responses.allResponses);
-      }
-    }, (resp: ErrorMessageOutput) => {
-      this.statusMessageService.showErrorToast(resp.error.message);
-    });
+    const missingRespMap: Map<string, ResponseOutput> = new Map();
+    const tmpMap: Map<string, ResponseOutput> = new Map();
+
+    if (this.hasSectionsLoadingFailed) {
+      // the page would not render properly
+      return;
+    }
+    of(...Object.keys(this.sectionsModel)).pipe(
+        concatMap((sectionName: string) => {
+          return of(this.feedbackSessionsService.getFeedbackSessionResults({
+            questionId,
+            courseId: this.session.courseId,
+            feedbackSessionName: this.session.feedbackSessionName,
+            intent: Intent.FULL_DETAIL,
+            groupBySection: sectionName,
+            sectionByGiverReceiver: 'giver',
+          }),
+          this.feedbackSessionsService.getFeedbackSessionResults({
+            questionId,
+            courseId: this.session.courseId,
+            feedbackSessionName: this.session.feedbackSessionName,
+            intent: Intent.FULL_DETAIL,
+            groupBySection: sectionName,
+            sectionByGiverReceiver: 'receiver',
+          }),
```

Comment:
There shouldn't be a need to split by giver/receiver here. There is a high likelihood for large amount of duplicates. Splitting by section is enough to give the needed performance boost.

It does mean that the back-end split of giver/receiver is not actually needed right now, but they can and will have their uses in other view types (GRQ and alike) eventually, so no work is wasted here.

Reply:
Updated to 'both'. Works as expected.